### PR TITLE
search function works, added product previews

### PR DESCRIPTION
**Pull request by cartershadd**
Product previews under the shopping page do show an error.

### CODE REVIEW

File: src/layout/HamburgerMenu.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+                    <span></span>
+                    <span></span>
+
+
+                    <ul className="menu">
+                        <li>
+                            <Link to="/cart" className="hamburgerLink"><i className="fas fa-shopping-cart"/></Link>
+                        </li>
+                        <li>
+                            <Link to="/" className="hamburgerLink">Home</Link>
```
Comment (nburn42) [inline src/layout/HamburgerMenu.js]: I don't see a route for this path in app.js

File: src/pages/ProductPreview.js [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+                    caption: "For your garden or kitchen.",
+                    category: "flower"
+                }
+            ],
+        };
+    }
+
+    render() {
+        return (
+            this.state.productList.filter(product => product.id === this.props.match.params.id).map((product, index) =>
```
Comment (nburn42) [inline src/pages/ProductPreview.js]: Nice

File: src/pages/Search.js [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+                    <input
+                        type="submit"
+                        value="Search"
+                        className="search-btn"
+                    />
+                </form>
+                <div className="searchResults">
+                    {this.state.filteredProductList.map((product, index) => {
+                        return <div key={index} className="product-profile">
+                            <Link to={`/${product.id}`} className="product-link">
```
Comment (nburn42) [inline src/pages/Search.js]: isn't the route `"/shop/:id"` not `/:id"`?

```<Route exact path="/cart" component={Cart}/>
                    <Route exact path="/shop" component={ShopPage}/>
                    <Route exact path="/shop/:id" component={ProductProfile}/>
                    <Route exact path="/search" component={Search}/>```

File: src/layout/HamburgerMenu.js (hunk 1 above)
Comment (cartershadd) [reply src/layout/HamburgerMenu.js]: Good point

File: src/pages/Search.js (hunk 3 above)
Comment (cartershadd) [reply src/pages/Search.js]: you are looking at the search component

**Pull request merged**